lms/djangoapps/ccx/views.py

```diff
@@ -181,6 +181,13 @@ def dashboard(request, course, ccx=None):
     else:
         context['create_ccx_url'] = reverse(
             'create_ccx', kwargs={'course_id': course.id})
+        if not run_extension_point(
+            'PCO_IS_LICENSED_AND_ADMIN',
+            user=request.user,
+            master_course=course.id,
+        ):
+            return HttpResponseForbidden(_('You must be a CCX coach to access this view.'))
```

Comment:
What do you think about moving this logic to `coach_dashboard` decorator and raising a 404 instead of showing a message?. The current behavior when CCX is disable is raising a 404.
```suggestion
        if not run_extension_point(
            'PCO_IS_LICENSED_AND_ADMIN',
            user=request.user,
            master_course=course.id,
        ):
            raise Http404
```

Reply:
When Http 404 is raised. If it is entered with the link directly and it is returned there is a django.core.urlresolvers problem. 
I also did the test moving the logic to the coach_dashboard decorator and the same problem exists.

![image](https://user-images.githubusercontent.com/30726391/182726874-8592dd00-7bbc-4696-914e-7d7587174834.png)